### ISSUE TITLE
Auto-completion by context

### ISSUE THREAD

**Issue by sbwoodside** [adamvoss/vscode-yaml#5]
Would it be possible to have auto-completion based on the YAML context. For example, if I'm typing a key in an associative array, it would complete based on other keys, if I'm typing a value, other values, etc?

**Comment by adamvoss** [comment]
Yes, it would be possible.  The [JSON implementation](https://github.com/Microsoft/vscode-json-languageservice/blob/master/src/services/jsonCompletion.ts) could be used as a reference for seeing how to interact with the Schema to get suggestions.  The [Language Server Protocol](https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md) document is probably the best source for documentation on how the completion API works to know what needs to be implemented.

**Comment by vojtechhabarta** [comment]
I know this is hard to implement because parser probably doesn't provide enough information while in the middle of editing but it would be very useful feature.

**Comment by JPinkney** [comment]
@sbwoodside @vojtechhabarta In our extension https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml we have this feature implemented if you want to try it out.
